Add proper `type` and refactored `cli` and `transform` functionalities

MahadMuhammad opened the pull request:
- Fixes https://github.com/Rust-GCC/rusttest-to-dg/issues/2 by using [`anyhow with_context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.with_context) method
- Updated the return type from `Vec<String>` to `String` 
- Refactored into each function, for parsing, converting source code to `String` variables, for removing and creating a file with updated code

MahadMuhammad committed:
Refactor code to use `anyhow::Context` for error handling

This will automatically print our custom `error message` and the underlying operating system error message.

Signed-off-by: Muhammad Mahad <mahadtxt@gmail.com>

MahadMuhammad committed:
Refactor transform_code into new file

- Updated the return type from `Vec<String>` to `String`

Signed-off-by: Muhammad Mahad <mahadtxt@gmail.com>

MahadMuhammad committed:
Refactor source code parsing and updating code into functions

- For parsing and converting source code to `String` variables, refactored into separate functions
- For removing and creating a file with updated code, refactored into separate function

Signed-off-by: Muhammad Mahad <mahadtxt@gmail.com>

CohenArthur commented on src/cli.rs:
```diff
@@ -25,6 +27,36 @@ pub struct Arguments {
+}
+
+pub fn update_source_code(args: &Arguments, new_code: String) -> Result<()> {
+    fs::remove_file(&args.source_file)
+        .with_context(|| format!("could not remove file `{}`", &args.source_file.display()))?;
+
+    fs::write(&args.source_file, new_code).with_context(|| {
+        format!(
+            "could not write updated code to file `{}`",
+            &args.source_file.display()
+        )
+    })?;
```
I don't think we want to remove the file and then create a new file at the same location. I think it's a bit confusing if you're using a tool, and it silently replaces your file :) plus, if you are working on the tool and trying it out on a Rust file, it's annoying that this file gets replaced everytime you run the tool. what if you make a mistake and you end up leaving a blank file?

I think a better solution would be to simply output the result on `stdout`, with `println!()` for example. so literally `println!("{new_code}")`

this way, you can run the tool like so:

```
./rusttest-to-dg -f file.rs -e file.stderr > new_file.rs
```

CohenArthur reviewed (commented):
this is a good cleanup and the code is looking good :D well done!

MahadMuhammad committed:
Redirect updated code to `stdout`

Instead of replacing the contents of file
and removed the code to removing and creating
a new file

Signed-off-by: Muhammad Mahad <mahadtxt@gmail.com>

CohenArthur commented on src/cli.rs:
```diff
@@ -43,6 +43,11 @@ pub fn parse_arguments_and_read_file(args: &Arguments) -> Result<(String, Option
+    println!("{source_code}");
+}
+
+#[allow(dead_code)]
```
can you delete the function instead of keeping it with `#[allow(dead_code)]`? it would be cleaner :)

CohenArthur reviewed (approved):
last nitpick but this is good to merge! great job!

CohenArthur reviewed (approved):
Thanks! LGTM!
